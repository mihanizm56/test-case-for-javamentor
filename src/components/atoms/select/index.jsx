import React from "react";
import Select from "@material-ui/core/Select";

export const renderSelect = ({ input, label, meta: { touched, error }, children, ...custom }) => (
	<Select fullWidth={true} {...input} children={children} {...custom} />
);
