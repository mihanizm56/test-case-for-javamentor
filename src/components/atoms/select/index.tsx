import React, { memo } from "react";
import SelectMaterial,{SelectProps as SelectPropsMaterial} from "@material-ui/core/Select";
import icon from "../../../assets/icons/scroll-icon.svg";



interface SelectProps extends SelectPropsMaterial {
  handleChange:(params: any)=>void, 
  selectedValue:string, 
  children?:Array<React.ReactNode>
}

export const Select = memo<SelectProps>(
  ({ handleChange, selectedValue='', children }) =>
    (
      <SelectMaterial
        fullWidth
        value={selectedValue}
        children={children}
        onChange={handleChange}
        style={{
          height: "40px",
          padding: "0 5px 0 10px",
          border: "1px solid blue",
          borderRadius: "10px"
        }}
        disableUnderline
        variant="outlined"
      />
    )
);
