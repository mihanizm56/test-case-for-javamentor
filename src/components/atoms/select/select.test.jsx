import React from "react";
import { shallow } from "enzyme";
import { Select } from "../select";
import { Button } from "../select";

describe("test Select", () => {
  const selectProps = {
    selectedValue: "test",
    handleChange: jest.fn()
  };
  let component;

  beforeEach(() => {
    jest.clearAllMocks();
    component = shallow(<Select {...selectProps} />);
  });

  describe("test render contract", () => {
    it("renders html-content", () => {
      expect(
        component
          .dive()
          .dive()
          .dive()
      ).toMatchSnapshot();
    });

    it("renders Select", () => {
      expect(
        component
          .dive()
          .dive()
          .dive()
          .find("Select").length
      ).toBe(1);
    });
  });

  describe("test props contract", () => {
    it("check handleClick prop", () => {
      expect(
        component
          .dive()
          .dive()
          .dive()
          .find("Select")
          .prop("onChange")
      ).toEqual(selectProps.handleChange);
    });
    it("check value of selectedValue prop was provided", () => {
      expect(
        component
          .dive()
          .dive()
          .dive()
          .find("Select")
          .prop("value")
      ).toEqual(selectProps.selectedValue);
    });
    it("check value of selectedValue prop was provided", () => {
      expect(
        component
          .dive()
          .dive()
          .dive()
          .find("Select")
          .prop("value")
      ).toEqual(selectProps.selectedValue);
    });
  });

  describe("test action contract", () => {
    it("calls prop handleChange function when change event was fired", () => {
      const testEvent = { target: { value: "42" } };

      component
        .dive()
        .dive()
        .dive()
        .find("Select")
        .simulate("change", testEvent);

      expect(selectProps.handleChange).toHaveBeenCalledWith(testEvent);
      expect(selectProps.handleChange).toHaveBeenCalledTimes(1);
    });
  });
});
