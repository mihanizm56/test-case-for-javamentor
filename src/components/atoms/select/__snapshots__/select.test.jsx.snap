// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Select test render contract renders html-content 1`] = `
<Select
  IconComponent={[Function]}
  autoWidth={false}
  classes={
    Object {
      "disabled": "MuiSelect-disabled-6",
      "filled": "MuiSelect-filled-3",
      "icon": "MuiSelect-icon-7",
      "outlined": "MuiSelect-outlined-4",
      "root": "MuiSelect-root-1",
      "select": "MuiSelect-select-2",
      "selectMenu": "MuiSelect-selectMenu-5",
    }
  }
  disableUnderline={true}
  displayEmpty={false}
  fullWidth={true}
  input={<WithStyles(Input) />}
  multiple={false}
  native={false}
  onChange={[MockFunction]}
  style={
    Object {
      "border": "1px solid blue",
      "borderRadius": "10px",
      "height": "40px",
      "padding": "0 5px 0 10px",
    }
  }
  value="test"
  variant="outlined"
/>
`;
